frontend/src/queries/nodes/DataNodeQuery.tsx

```diff
@@ -0,0 +1,41 @@
+import MonacoEditor from '@monaco-editor/react'
+import { useEffect, useState } from 'react'
+import { query as runQuery } from '../query'
+import { AutoSizer } from 'react-virtualized/dist/es/AutoSizer'
+import { Node } from '~/queries/schema'
+
+interface DataNodeQueryProps {
+    query: Node
+}
+
+/** Default renderer for data nodes. Display the JSON in a Monaco editor.  */
+export function DataNodeQuery({ query }: DataNodeQueryProps): JSX.Element {
+    const [response, setResponse] = useState<string>('Loading...')
+
+    useEffect(() => {
+        // The simplest thing for now.
+        async function fetchIt(): Promise<void> {
+            try {
+                const resp = await runQuery(query)
+                setResponse(JSON.stringify(resp, null, 2))
+            } catch (e: any) {
+                setResponse(`Error: ${e.message}`)
+            }
+        }
+        void fetchIt()
+    }, [JSON.stringify(query)])
```

Comment:
Could this be just `query`?

Reply:
Nope, as we don't know where the object comes from, and it might be an inline object that's recreated on every render 

```ts
function RenderStuff() {
  return <Query query={{ kind: 'YOLO' }} />
}
```

In this case the `props.query !== nextProps.query`